pkg/ddc/alluxio/dataset.go

```diff
@@ -152,7 +151,92 @@ func (e *AlluxioEngine) UpdateDatasetStatus(phase datav1alpha1.DatasetPhase) (er
 // func (e *AlluxioEngine) IsBoundToDataset() (bound bool, err error) {
 // 	return
 // }
+func (e *AlluxioEngine) UpdateMountStatus(phase datav1alpha1.DatasetPhase) (err error) {
+	// 1. update the runtime status
+	runtime, err := e.getRuntime()
+	if err != nil {
+		return err
+	}
+
+	// 2.update the dataset status
+	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		dataset, err := utils.GetDataset(e.Client, e.name, e.namespace)
+		if err != nil {
+			return err
+		}
+		datasetToUpdate := dataset.DeepCopy()
+		var cond datav1alpha1.DatasetCondition
+
+		switch phase {
+		case datav1alpha1.UpdatingDatasetPhase:
+			cond = utils.NewDatasetCondition(datav1alpha1.DatasetUpdating, datav1alpha1.DatasetUpdatingReason,
+				"The ddc runtime is updating.",
+				corev1.ConditionTrue)
+		case datav1alpha1.UpdatedDatasetPhase:
+			cond = utils.NewDatasetCondition(datav1alpha1.DatasetUpdateReady, datav1alpha1.DatasetUpdateReadyReason,
+				"The ddc runtime is updated.",
+				corev1.ConditionTrue)
+		default:
+			cond = utils.NewDatasetCondition(datav1alpha1.DatasetReady, datav1alpha1.DatasetReadyReason,
+				"The ddc runtime is unknown.",
+				corev1.ConditionFalse)
+		}
+
+		datasetToUpdate.Status.Phase = phase
+		datasetToUpdate.Status.Conditions = utils.UpdateDatasetCondition(datasetToUpdate.Status.Conditions,
+			cond)
+
+		datasetToUpdate.Status.CacheStates = runtime.Status.CacheStates
+		// datasetToUpdate.Status.CacheStates =
+
+		if datasetToUpdate.Status.HCFSStatus == nil {
```

Comment:
I think We only need to update mountStatus, condition and status of the dataset. Right?